packages/shared/src/utils.ts

```diff
@@ -47,17 +52,21 @@ export function getValues(arr: AnyNumArray): NumArray {
   return isNdArray(arr) ? arr.data : arr;
 }
 
-export function toTypedNdArray<
-  T extends NumArray,
-  U extends TypedArrayConstructor
->(arr: NdArray<T>, TypedArrayConstructor: U) {
-  if (isTypedNdArray(arr)) {
+export function toTypedNdArray<T extends TypedArrayConstructor>(
+  arr: NdArray<NumArray>,
+  Constructor: T
+): NdArray<InstanceType<T>> {
+  return ndarray(Constructor.from(arr.data) as InstanceType<T>, arr.shape);
```

Comment:
Just renamed an argument and moved the type assertion closer to where it needs to be.

Reply:
Rereading this again, I wonder: does it makes a copy of the data or not ?

Reply:
Yes, definitely. That's why I was hoping to be able to do the BGR flip in the shader at some point. But the use in `applyMapping` is worth a look as well, as I'm just realising that we always call `createArrayFromView` from there, regardless of whether the base array is sliced/mapped...

Reply:
I've opened issue #961